minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/MinikubeExtension.java

```diff
@@ -16,16 +16,32 @@
 
 package com.google.cloud.tools.minikube;
 
+import com.google.cloud.tools.minikube.util.CommandExecutorFactory;
+import com.google.cloud.tools.minikube.util.MinikubeDockerEnvParser;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
 import org.gradle.api.Project;
 import org.gradle.api.provider.PropertyState;
 
 /** Minikube configuration extension. */
 public class MinikubeExtension {
   private final PropertyState<String> minikube;
 
+  // @VisibleForTesting
+  MinikubeExtension setCommandExecutorFactory(CommandExecutorFactory commandExecutorFactory) {
```

Comment:
Oops, I think you can pass this through the constructor instead of here.

Reply:
```
public MinikubeExtension(Project project, CommandExecutorFactory factory)
```

and then in MinikubePlugin.java when you create the extension, you just pass in a command executor factory, and that same factory can also be set on the Task in MinikubePlugin.java

Reply:
So to add it to the MinikubeTask, we would need a MinikubeTask#setCommandExecutorFactory. Is this what we want?

Reply:
We would need to add a MinikubeTask#setCommandExecutorFactory method to pass the plugin's command executor factory into MinikubeTask. Is this what we want?